version: '3.8'

services:
  postgresql:
    image: postgres:16.2
    container_name: postgresql
    restart: "always"
    env_file:
      - .env
    ports:
      - 5432:5432
    volumes:
      - ./data:/var/lib/postgresql/data
      - ./postgresql.conf:/etc/postgresql/postgresql.conf
      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "${POSTGRES_USER}"]
      interval: 10s
      retries: 5
      start_period: 5s
    networks:
      - devops-network

  pgadmin:
    image: dpage/pgadmin4:latest
    container_name: pgadmin
    restart: "always"
    env_file:
      - .env
    ports:
      - 5433:80
    volumes:
      - ./pgadmin:/var/lib/pgadmin
    networks:
      - devops-network
    depends_on:
      postgresql:
        condition: service_healthy

networks:
  devops-network:
    external: true
